### ISSUE TITLE
Add token based authentication

### ISSUE THREAD

**Issue by KacperKluka** [ably/ably-asset-tracking-swift#134]
Provide the capability to authenticate users using Token Authentication. 
See [Android Issue](https://github.com/ably/ably-asset-tracking-android/issues/171) and [Android PR](https://github.com/ably/ably-asset-tracking-android/pull/222) for more details.

Acceptance criteria: 
- token request approach implemented

┆Issue is synchronized with this [Jira Story](https://ably.atlassian.net/browse/SDK-743) by [Unito](https://www.unito.io)

**Comment by ben-xD** [comment]
I've just gotten to the point of writing tests, and needed to make a new Xcode project which depends on both Subscriber and Publisher SDKs, so I can do integration tests (following Android library as an example). To avoid manually adding every file being tested into this project (as currently being done in this project), I wanted to create a separate **module** to import. This proved very difficult with Cocoapods (and finally, not worth continuing because Cocoapods will be mostly removed once SPM is supported), so I will just manually add these files to "Compile Sources..." until we add SPM support: https://github.com/ably/ably-asset-tracking-cocoa/issues/148 

A similar question to mine (unanswered to Stack Overflow) is https://stackoverflow.com/questions/53314282/how-do-you-use-one-subspec-as-the-dependency-of-another-subspec-with-cocoapods

TLDR: Using subspecs is tedious and uncommon (current method) and I look forward to removing this and using Swift Package Manager.

**Comment by ben-xD** [comment]
It also looks like Cocoapods was actually limiting the usage of both SDK in the same target. Because the SDKs are compiled into the same target (AblyAssetTracking, e.g. `import AblyAssetTracking`), the existing code would not compile, because it **redeclares the same symbols** separately in the Publisher and Subscriber SDKs. Have a look at these warnings: (on the left is publisher, on the right is subscriber):

![Screenshot 2021-06-10 at 17 03 55](https://user-images.githubusercontent.com/24711048/121558774-d9a84000-ca0d-11eb-82ed-18e023b9cda2.png)

This will also be fixed by moving to SPM, as these libraries will be different targets, Publisher and Subscriber SDKs and will be in a separate namespace. To fix this in Cocoapods, 3 separate Podspec files are needed (Publisher, Subscriber, and Core).
